cpp/src/lance/arrow/dataset.cc

```diff
@@ -46,23 +48,22 @@ namespace {
 /// \param base_uri the base uri of the dataset
 /// \param version optional version number. If not specified, returns the latest version.
 /// \return string path to the manifest file.
-std::string GetManifestPath(const std::string& base_uri, std::optional<uint64_t> version) {
+std::string GetManifestPath(const std::string& base_uri,
+                            std::optional<uint64_t> version = std::nullopt) {
   if (version.has_value()) {
     return fs::path(base_uri) / kVersionsDir / fmt::format("{}.manifest", version.value());
   } else {
     return fs::path(base_uri) / kLatestManifest;
   }
 }
 
-std::vector<std::shared_ptr<lance::format::DataFragment>> CreateFragments(
-    const std::vector<std::string>& paths, const lance::format::Schema& schema) {
-  std::vector<std::shared_ptr<lance::format::DataFragment>> fragments;
+auto CreateFragments(const std::vector<std::string>& paths, const lance::format::Schema& schema) {
   auto field_ids = schema.GetFieldIds();
-  for (auto path : paths) {
-    fragments.emplace_back(
-        std::make_shared<lance::format::DataFragment>(lance::format::DataFile(path, field_ids)));
-  }
-  return fragments;
+  return paths | views::transform([&field_ids](auto& path) {
```

Comment:
just confirming: these don't need any error handling because they're not actually opening the resources right?

Reply:
Yea, there is no system call in this transform.